### PR TITLE
Implement draft-16

### PR DESCRIPTION
**Pull request by mad-p**
Re: #25, #26 

draft-15 changes
* Add http_1_1_required error code

draft-16 changes
* PRIORITY can be sent/received in any state
* Update hpack-test-case to lastest (though no normative changes included)
* Examples advertise h2-16

Bugs found in added tests are fixed, too.

### CODE REVIEW

File: lib/http/2/stream.rb [hunk 1]
```diff
@@ -98,14 +99,14 @@ def receive(frame)
       case frame[:type]
       when :data
         # TODO: when receiving DATA, keep track of local_window.
```
Comment (igrigorik) [inline lib/http/2/stream.rb]: we can drop the TODO now, correct?

File: lib/http/2/stream.rb [hunk 2]
```diff
@@ -98,14 +99,14 @@ def receive(frame)
...
+        @local_window -= frame[:payload].size
         emit(:data, frame[:payload]) if !frame[:ignore]
       when :headers, :push_promise
         emit(:headers, frame[:payload]) if !frame[:ignore]
       when :priority
         process_priority(frame)
       when :window_update
-        @remote_window += frame[:increment]
-        send_data
+        process_window_update(frame) if !frame[:ignore]
```
Comment (igrigorik) [inline lib/http/2/stream.rb]: Is the separate method call necessary? Seems like we're only calling this method from just this one place? Also the if check is duplicated.

File: lib/http/2/stream.rb (hunk 1 above)
Comment (mad-p) [inline lib/http/2/stream.rb]: fixed

File: lib/http/2/stream.rb (hunk 2 above)
Comment (mad-p) [inline lib/http/2/stream.rb]: Intended to be called from connection.
Just removed duplicate :ignore check.

**Comment by igrigorik** [comment]
awesome work, as usual! :+1:

**Pull request merged**